plugin/evm/atomic/sync/atomic_sync_extender.go

```diff
@@ -0,0 +1,79 @@
+// (c) 2021-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	syncclient "github.com/ava-labs/coreth/sync/client"
+
+	"github.com/ava-labs/coreth/plugin/evm/atomic/state"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+	"github.com/ava-labs/coreth/plugin/evm/sync"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var _ sync.Extender = (*AtomicSyncExtender)(nil)
+
+type AtomicSyncExtender struct {
+	backend              state.AtomicBackend
+	stateSyncRequestSize uint16
+}
+
+func NewAtomicSyncExtender(backend state.AtomicBackend, stateSyncRequestSize uint16) *AtomicSyncExtender {
+	return &AtomicSyncExtender{
+		backend:              backend,
+		stateSyncRequestSize: stateSyncRequestSize,
+	}
+}
+
+func (a *AtomicSyncExtender) Sync(ctx context.Context, client syncclient.Client, verDB *versiondb.Database, syncSummary message.Syncable) error {
```

Comment:
`client syncclient.Client` can be the narrower interface `client syncclient.LeafClient` (`Extender` interface needs an update).